ui/src/configuration/components/Telegrafs.tsx

```diff
@@ -0,0 +1,352 @@
+          onDismiss={this.handleCloseInstructions}
+        />
+        <TelegrafConfigOverlay
+          visible={this.isTelegrafConfigVisible}
+          onDismiss={this.handleCloseTelegrafConfig}
+        />
+      </>
+    )
+  }
+
+  private get hasNoBuckets(): boolean {
+    const {buckets} = this.props
+
+    if (!buckets || !buckets.length) {
+      return true
+    }
+
+    return false
+  }
+
+  private get telegrafsWizard(): JSX.Element {
+    const {buckets} = this.props
+
+    if (this.hasNoBuckets) {
+      return
+    }
+
+    return (
+      <CollectorsWizard
+        visible={this.isDataLoaderVisible}
+        onCompleteSetup={this.handleDismissDataLoaders}
+        startingStep={0}
+        buckets={buckets}
+      />
+    )
+  }
+
+  private get selectedCollector() {
+    return this.props.telegrafs.find(c => c.id === this.state.collectorID)
+  }
+
+  private get isDataLoaderVisible(): boolean {
+    return this.state.dataLoaderOverlay === OverlayState.Open
+  }
+
+  private get isInstructionsVisible(): boolean {
+    return this.state.instructionsOverlay === OverlayState.Open
+  }
+
+  private handleOpenInstructions = (collectorID: string): void => {
+    this.setState({
+      instructionsOverlay: OverlayState.Open,
+      collectorID,
+    })
+  }
+
+  private handleCloseInstructions = (): void => {
+    this.setState({
+      instructionsOverlay: OverlayState.Closed,
+      collectorID: null,
+    })
+  }
+
+  private get isTelegrafConfigVisible(): boolean {
+    return this.state.telegrafConfig === OverlayState.Open
+  }
+
+  private handleOpenTelegrafConfig = (
+    telegrafID: string,
+    telegrafName: string
+  ): void => {
+    this.props.onSetTelegrafConfigID(telegrafID)
+    this.props.onSetTelegrafConfigName(telegrafName)
+    this.setState({
+      telegrafConfig: OverlayState.Open,
+    })
+  }
+
+  private handleCloseTelegrafConfig = (): void => {
+    this.props.onClearDataLoaders()
+    this.setState({
+      telegrafConfig: OverlayState.Closed,
+    })
+  }
+
+  private get createButton(): JSX.Element {
+    let status = ComponentStatus.Default
+    let titleText = 'Create a new Telegraf Configuration'
+
+    if (this.hasNoBuckets) {
+      status = ComponentStatus.Disabled
+      titleText =
+        'You need at least 1 bucket in order to create a Telegraf Configuration'
+    }
+
+    return (
+      <Button
+        text="Create Configuration"
+        icon={IconFont.Plus}
+        color={ComponentColor.Primary}
+        onClick={this.handleAddCollector}
+        status={status}
+        titleText={titleText}
+      />
+    )
+  }
+
+  private handleAddCollector = () => {
+    const {buckets, onSetBucketInfo, onSetDataLoadersType} = this.props
+
+    if (buckets && buckets.length) {
+      const {organization, organizationID, name, id} = buckets[0]
+      onSetBucketInfo(organization, organizationID, name, id)
+    }
+
+    onSetDataLoadersType(DataLoaderType.Scraping)
```

Comment:
shouldnt this type be `streaming`? or (collectors/telegraf) whatever its been named